### PR TITLE
SDL2/EGL: handle GL context attributes (#7227)

### PR DESCRIPTION
**Pull request by Beuc**
Cf. #7227

### CODE REVIEW

File: src/library_egl.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ var LibraryEGL = {
     currentContext: 0 /* EGL_NO_CONTEXT */,
     currentReadSurface: 0 /* EGL_NO_SURFACE */,
     currentDrawSurface: 0 /* EGL_NO_SURFACE */,
+    alpha: false,
+    depth: true,
+    stencil: true,
+    antialias: true,
```
Comment (kripken) [inline src/library_egl.js]: Are these defaults set according to EGL's defaults, or WebGL, or just based on the previous defaults we had here? Please add a comment on this, as e.g. the eglChooseConfig docs mention depth and stencil should be false, so we should document the difference there.

File: src/library_egl.js [hunk 2]
```diff
@@ -35,7 +39,31 @@ var LibraryEGL = {
...
         return 0;
       }
-      // TODO: read attribList.
+
+      // read attribList.
+      for(;;) {
+        var param = {{{ makeGetValue('attribList', '0', 'i32') }}};
+        if (param == 0x3021 /*EGL_ALPHA_SIZE*/) {
+          var alphaSize = {{{ makeGetValue('attribList', '4', 'i32') }}};
+	  EGL.alpha = (alphaSize > 0);
```
Comment (kripken) [inline src/library_egl.js]: indentation looks wrong here

File: tests/test_browser.py [hunk 3]
```diff
@@ -1247,6 +1247,7 @@ def test_webgl_context_attributes(self):
     # perform tests with attributes activated
     self.btest('test_webgl_context_attributes_glut.c', '1', args=['--js-library', 'check_webgl_attributes_support.js', '-DAA_ACTIVATED', '-DDEPTH_ACTIVATED', '-DSTENCIL_ACTIVATED', '-DALPHA_ACTIVATED', '-lGL', '-lglut', '-lGLEW'])
     self.btest('test_webgl_context_attributes_sdl.c', '1', args=['--js-library', 'check_webgl_attributes_support.js', '-DAA_ACTIVATED', '-DDEPTH_ACTIVATED', '-DSTENCIL_ACTIVATED', '-DALPHA_ACTIVATED', '-lGL', '-lSDL', '-lGLEW'])
+    self.btest('test_webgl_context_attributes_sdl2.c', '1', args=['--js-library', 'check_webgl_attributes_support.js', '-DAA_ACTIVATED', '-DDEPTH_ACTIVATED', '-DSTENCIL_ACTIVATED', '-DALPHA_ACTIVATED', '-lGL', '-s', 'USE_SDL=2', '-lGLEW'])
```
Comment (kripken) [inline tests/test_browser.py]: is that a new file? maybe forgot to add it to the PR. but also, should it have `egl` in the name?

File: src/library_egl.js (hunk 1 above)
Comment (Beuc) [reply src/library_egl.js]: Right, I just avoided changing the previous behavior, I'll add a comment.
Would you rather we follow EGL's defaults though?

File: tests/test_browser.py (hunk 3 above)
Comment (Beuc) [reply tests/test_browser.py]: -_-'
That's a new file. I decided to run the test at the SDL2 level, which is what people more commonly use, and which itself relies on EGL.

File: src/library_egl.js (hunk 1 above)
Comment (kripken) [reply src/library_egl.js]: Let's leave that for a later PR - I'm not sure what's the best thing.

File: tests/test_browser.py (hunk 3 above)
Comment (kripken) [reply tests/test_browser.py]: Oh, I see, so this PR fixes SDL2 as well?

**Comment by kripken** [comment]
Thanks! Everything looks good, however `browser.test_sdl2glshader` fails both in the firefox and other modes. Does it fail for you locally too? Let me know if the problem isn't obvious to you, it could be a weird closure issue that I can look into.

File: tests/test_browser.py (hunk 3 above)
Comment (Beuc) [reply tests/test_browser.py]: Yes, that's the goal :)

**Comment by Beuc** [comment]
I introduced a new variable without "var" and Closure complained.
It took longer to locate the error in closure's extremely verbose output than to fix ;)

**Comment by kripken** [comment]
Great, thanks!

**Pull request merged**